### PR TITLE
Expand on contracting carve-out rule test

### PR DESCRIPTION
**Pull request by instagibbs**
Testing the "one more" on top of "one more", making sure this is rejected.

### CODE REVIEW

File: test/functional/mempool_package_onemore.py [hunk 1]
```diff
@@ -48,10 +48,17 @@ def run_test(self):
...
+        assert_equal(len(self.nodes[0].getrawmempool(True)), 0)
+
         fee = Decimal("0.0002")
         # MAX_ANCESTORS transactions off a confirmed tx should be fine
         chain = []
-        for _ in range(4):
+        # First tx has 3 outputs
+        (txid, sent_value) = self.chain_transaction(self.nodes[0], [txid], [vout], value, fee, 3)
+        vout = 0
+        value = sent_value
```
Comment (jonatack) [inline test/functional/mempool_package_onemore.py:59]: redundant assignment in this loop (and the next one), can remove?

if not, may make sense to extract the loop body to a function

File: test/functional/mempool_package_onemore.py [hunk 2]
```diff
@@ -48,10 +48,17 @@ def run_test(self):
         vout = utxo[0]['vout']
         value = utxo[0]['amount']
 
+        assert_equal(len(self.nodes[0].getrawmempool(True)), 0)
+
         fee = Decimal("0.0002")
         # MAX_ANCESTORS transactions off a confirmed tx should be fine
         chain = []
-        for _ in range(4):
+        # First tx has 3 outputs
```
Comment (jonatack) [inline test/functional/mempool_package_onemore.py:56]: verified, this is correct, first tx has 3 outputs and the next three txs have 2 outputs

File: test/functional/mempool_package_onemore.py [hunk 3]
```diff
@@ -79,6 +86,10 @@ def run_test(self):
         # and the second chain should work just fine
         self.chain_transaction(self.nodes[0], [second_chain], [0], second_chain_value, fee, 1)
 
+        # But now that we're at the additional carve-out limit, the third base output being spent will be rejected
+        # For both ancestor's descendant limits
+        assert_raises_rpc_error(-26, "too-long-mempool-chain, too many descendants", self.chain_transaction, self.nodes[0], [chain[0][0]], [2], chain[0][1], fee, 1)
```
Comment (jonatack) [inline test/functional/mempool_package_onemore.py:91]: "too-long-mempool-chain, too many descendants" used 5x, could hoist to a `TOO_MANY_DESCENDANTS` constant

File: test/functional/mempool_package_onemore.py [hunk 4]
```diff
@@ -48,10 +48,17 @@ def run_test(self):
         vout = utxo[0]['vout']
         value = utxo[0]['amount']
 
+        assert_equal(len(self.nodes[0].getrawmempool(True)), 0)
```
Comment (jonatack) [inline test/functional/mempool_package_onemore.py:51]: perhaps add a comment why

**Review by jonatack — COMMENTED** [review]
ACK c5db55a6a00918ed74c9fd modulo comments (feel free to ignore the nits)

File: test/functional/mempool_package_onemore.py (hunk 2 above)
Comment (instagibbs) [reply test/functional/mempool_package_onemore.py:56]: heh I supposed this should just be an assert

File: test/functional/mempool_package_onemore.py (hunk 4 above)
Comment (instagibbs) [reply test/functional/mempool_package_onemore.py:51]: done

File: test/functional/mempool_package_onemore.py (hunk 3 above)
Comment (instagibbs) [reply test/functional/mempool_package_onemore.py:91]: done

File: test/functional/mempool_package_onemore.py [hunk 5]
```diff
@@ -48,10 +48,17 @@ def run_test(self):
...
+
         fee = Decimal("0.0002")
         # MAX_ANCESTORS transactions off a confirmed tx should be fine
         chain = []
-        for _ in range(4):
+        # First tx has 3 outputs
+        (txid, sent_value) = self.chain_transaction(self.nodes[0], [txid], [vout], value, fee, 3)
+        vout = 0
```
[the remark this answers is not in the document]
Comment (instagibbs) [reply test/functional/mempool_package_onemore.py:60]: removed for line 63, otherwise I think this is more clear personally

**Pull request closed without merging**